dds/DCPS/Serializer.inl

```diff
@@ -1315,16 +1315,22 @@ ACE_INLINE bool
 operator>>(Serializer& s, ACE_InputCDR::to_string x)
 {
   const size_t length = s.read_string(const_cast<char*&>(x.val_));
-  return s.good_bit()
-         && ((x.bound_ == 0) || (length <= x.bound_));
+  if ((x.bound_ != 0) && (length > x.bound_)) {
+    s.set_construction_status(Serializer::BoundConstructionFailure);
+    return false;
+  }
+  return s.good_bit();
```

Comment:
In the previous version, `length` is not used when `read_string` fails (detected via good_bit).